backend/src/core/objects/files/nodeComposer.ts

```diff
@@ -44,7 +44,7 @@ export const composeNodesDataAsFileReadable = async ({
         )
 
         if (chunkedData.some((e) => e === undefined)) {
-          const notFoundChunkIndex = chunksToDownload.findIndex(
+          const notFoundChunkIndex = chunkedData.findIndex(
```

Comment:
This was a check that we were performing incorrectly by checking if the requested cid was not undefined rather than the actual data of that CID